build-index/tagclient/client.go

```diff
@@ -38,6 +38,7 @@ var (
 
 // Client wraps tagserver endpoints.
 type Client interface {
+	CheckReadiness() error
```

Comment:
My understanding is the readiness endpoint is provided to JC control plane which will call it through http. So it is not necessary to add a client for the endpoint.

Reply:
The agent's readiness endpoint will check the build index' readiness, as build index is a dependency for agent. I'll add that context to the description.